src/frontend/pages/Products/components/FiltersListing.jsx

```diff
@@ -0,0 +1,184 @@
+import React from "react";
+import { useProducts } from "../../../contexts/products-context";
+
+function Checkbox({ checked, label, changeType }) {
+  const { dispatch } = useProducts();
+
+  return (
+    <div className="pb-05">
+      <input
+        type="checkbox"
+        id={label}
+        value={label}
+        checked={checked}
+        onChange={() => dispatch({ type: changeType })}
+      />
+      <label className="p-1" htmlFor="scales">
+        {label}
+      </label>
+    </div>
+  );
+}
+
+function RadioButton({ checkedType, label }) {
+  const { state, dispatch } = useProducts();
+  return (
+    <div className="pb-05">
+      <input
+        type="radio"
+        id={label}
+        value={label}
+        name="rating"
+        checked={state.filterBy === checkedType}
+        onChange={() => dispatch({ type: checkedType })}
+      />
+      <label className="p-1" htmlFor={label}>
+        {label}
+      </label>
+    </div>
+  );
+}
```

Comment:
You can keep these functions in a separate file and import it here. This will make the FiltersListing component clean and short.